tools/vms-generator/utils/utils.go

```diff
@@ -1195,6 +1196,19 @@ func GetVMIWithHookSidecar() *v1.VirtualMachineInstance {
 	return vmi
 }
 
+func GetVMIWithGenericHook() *v1.VirtualMachineInstance {
+	vmi := getBaseVMI(VmiWithHookGeneric)
+	vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("1024M")
+
+	initFedora(&vmi.Spec)
+	addNoCloudDiskWitUserData(&vmi.Spec, generateCloudConfigString(cloudConfigUserPassword))
+
+	vmi.ObjectMeta.Annotations = map[string]string{
+		"hooks.kubevirt.io/hookSidecars": fmt.Sprintf("[{\"args\": [\"--version\", \"v1alpha2\"], \"image\": \"%s/sidecar-shim:%s\", \"configMap\": {\"name\": \"my-config-map\", \"key\": \"my_script.sh\"}}]", DockerPrefix, DockerTag),
```

Comment:
Let's break this 234 chars long line. I'd `fmt.Sprintf` to a local variable. Using `` raw literals should be fine here to avoid escaping "

Reply:
I spent an unreasonable amount of time on this one. Marshalling to json and then using its string form didn't work out. There always used to be a missing quote `'` that made the yaml incorrect. So I have gone with the raw literals.

Reply:
I guess we can always refactor later.